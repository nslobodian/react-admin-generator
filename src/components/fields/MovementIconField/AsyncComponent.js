import React, { Component } from 'react'
import connect from 'react-redux/lib/connect/connect'
import { getMovementIconsAttempt } from 'redux/NotificationTag'
import MovementIconField from './MovementIconField'
import { injectSagas } from 'store/sagas'
import { getStore } from 'store/createStore'

export class CreateNewAudienceWrapper extends Component {
  componentDidMount () {
    import(/* webpackChunkName: 'Component MovementIconField' */ './getUserMovementsIcons')
      .then(({ getUserMovementsIcons }) => {
        injectSagas(getStore(), { key: 'ComponentMovementIconField', sagas: [getUserMovementsIcons] })
        this.props.getMovementIconsAttempt(this.props.notifications)
      })
  }

  render () {
    return <MovementIconField {...this.props} />
  }
}

const mapActionCreators = { getMovementIconsAttempt }

export default connect(({ NotificationTag: { movementIcons } }) => ({ movementIcons }), mapActionCreators)(CreateNewAudienceWrapper)

