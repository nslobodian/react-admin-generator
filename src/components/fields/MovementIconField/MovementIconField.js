import React from 'react'
import Field from 'redux-form/lib/Field'
import DropdownField from 'components/Dropdown/DropdownElement'

const MovementIconField = ({ movementIcons, name = 'icon', placeholder = 'Choose An Icon', label = 'Icon', notifications }) =>
  <div className='form-group'>
    <label htmlFor={name} className='text-muted'>{label}</label>
    <Field
      name={name}
      placeholder={placeholder}
      options={movementIcons}
      component={DropdownField}
      labelKey={notifications ? 'title' : 'pictureUrl'}
      valueKey='iconId'
      optionRenderer={notifications ? null : ({ pictureUrl, movementIconId }) => <img src={pictureUrl}
        alt={movementIconId} />}
      valueRenderer={notifications ? null : ({ pictureUrl }, movementIconId) => <img src={pictureUrl}
        alt={movementIconId} />}
    />
  </div>

export default MovementIconField
