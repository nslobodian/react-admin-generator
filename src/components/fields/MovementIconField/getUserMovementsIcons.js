import { call, put, take } from 'redux-saga/effects'
import request from 'sagas/api'
import {
  GET_MOVEMENTS_ICONS_ATTEMPT,
  GET_MOVEMENTS_ICONS_FAILURE,
  GET_MOVEMENTS_ICONS_SUCCESSFULLY,
} from 'redux/NotificationTag'
import { sagaErrorNotification } from 'utils/notifications'

export function* getUserMovementsIcons () {
  while (true) {
    const { notifications } = yield take(GET_MOVEMENTS_ICONS_ATTEMPT)
    const body = {
      method: 'GET',
    }

    try {
      const { icons } = yield call(request, `${notifications ? 'notifications' : 'movements'}/icons`, body)
      yield put({
        type: GET_MOVEMENTS_ICONS_SUCCESSFULLY,
        movementIcons: icons,
      })
    } catch (error) {
      yield put(sagaErrorNotification(GET_MOVEMENTS_ICONS_FAILURE))
    }
  }
}
