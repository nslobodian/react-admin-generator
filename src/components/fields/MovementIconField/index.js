import MovementIconField from './AsyncComponent'

export default MovementIconField
