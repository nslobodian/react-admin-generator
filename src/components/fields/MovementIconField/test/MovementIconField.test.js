import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import MovementIconField from '../MovementIconField'

describe('(Components) MovementIconField', () => {
  it('Should render component properly', () => {
    const props = {
      movementIcons: true,
      notifications: true,
    }
    const wrapper = shallow(
      <MovementIconField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('No notifications', () => {
    const props = {
      movementIcons: true,
      notifications: false,
      pictureUrl: 'pictureUrl',
      movementIconId: 1,
    }
    const wrapper = shallow(
      <MovementIconField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
