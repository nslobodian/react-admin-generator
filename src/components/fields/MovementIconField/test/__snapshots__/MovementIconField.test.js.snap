// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) MovementIconField No notifications 1`] = `
<div
  className="form-group"
>
  <label
    className="text-muted"
    htmlFor="icon"
  >
    Icon
  </label>
  <Field
    component={[Function]}
    labelKey="pictureUrl"
    name="icon"
    optionRenderer={[Function]}
    options={true}
    placeholder="Choose An Icon"
    valueKey="iconId"
    valueRenderer={[Function]}
  />
</div>
`;

exports[`(Components) MovementIconField Should render component properly 1`] = `
<div
  className="form-group"
>
  <label
    className="text-muted"
    htmlFor="icon"
  >
    Icon
  </label>
  <Field
    component={[Function]}
    labelKey="title"
    name="icon"
    optionRenderer={null}
    options={true}
    placeholder="Choose An Icon"
    valueKey="iconId"
    valueRenderer={null}
  />
</div>
`;
